project/npda/views/npda_users.py

```diff
@@ -563,21 +543,17 @@ def done(self, form_list, **kwargs):
         # this will not be called if debug=True
         response = super().done(form_list)
         response_url = getattr(response, "url")
-        # successful login, get PDU, ODS and organisation details from user and store in session
-        current_user_ods_code = (
-            self.request.user.organisation_employers.first().ods_code
+        request = self.request
+        
+        # Update the session with the new session object
+        # Get the organisation employer
+        organisation_employer = user.organisation_employers.first()
+
+        new_session_object = create_session_object_from_organisation_employer(
+            organisation_employer
         )
-        current_user_pz_code = self.request.user.organisation_employers.first().pz_code
-        if "ods_code" not in self.request.session:
-            self.request.session["ods_code"] = current_user_ods_code
-
-        if "pz_code" not in self.request.session:
-            self.request.session["pz_code"] = current_user_pz_code
 
-        if "sibling_organisations" not in self.request.session:
-            sibling_organisations = get_single_pdu_from_ods_code(current_user_ods_code)
-            # store the results in session
-            self.request.session["sibling_organisations"] = sibling_organisations
+        request.session.update(new_session_object)
```

Comment:
I'd just refer to `self.request` directly if we can as that matches elsewhere in the code